### PR TITLE
Strict and native Types + PHP8.0 Drop + PHPUnit Upgrade [BC Break]

### PR DESCRIPTION
**Pull request by marcelthole**
|    Q          |   A
|-------------- | ------
| BC Break      | yes

### Description

As mentioned in https://github.com/laminas/laminas-crypt/pull/34#issuecomment-2228283417 we could use the chance and add strict types and native types for the next release if we have already BC Breaks.

So here some (mostly automated) changes:
- use `declare(strict_types=1);
- Drop mcrypt support (was only for PHP 7)
- Drop PHP 8.0 support
- Upgrade PHPUnit to Version 10 and laminas-coding-standard to 2.5
- Use native typehints for every method and property

What is missing?
- use more strict array annotations, but these would not be a BC Break in the future
- Maybe also introduce psalm? But that's more like an feature so i didn't added it here

### CODE REVIEW

**Comment by Ocramius** [comment]
Note: we need to open `4.0.x` before going for a merge here

File: src/PublicKey/Rsa/PublicKey.php
Comment (gsteel) [inline src/PublicKey/Rsa/PublicKey.php:142]: This removal should probably be documented

**Review by gsteel — COMMENTED** [review]
Looks good to me - I can't see any problematic changes other than the obvious BC breaks with native types. I'd approve if I was more familiar with this component, but I'll leave that to other members of the TSC 👍

File: src/PublicKey/Rsa/PublicKey.php
Comment (marcelthole) [reply src/PublicKey/Rsa/PublicKey.php:142]: is there something like a changelog or will it be manually filled with a release of a new tag?

Comment (gsteel) [reply src/PublicKey/Rsa/PublicKey.php:142]: Just searched - There's no mention of it in any docs or any usage anywhere else, not even in tests.

As it's not mentioned in the docs, perhaps all that is required is a mention in the release notes. I don't think this is a big deal👍

Comment (boesing) [reply src/PublicKey/Rsa/PublicKey.php:142]: I tend to add stuff like this to the milestone manually using keep-a-changelog format.
did that a bunch of times for laminas-cache and related adapters.

https://github.com/laminas/laminas-cache/releases/tag/4.0.0

**Comment by Xerkus** [comment]
This component was voted to be abandoned but we failed to follow up and mark it as such. https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2023-11-06-TSC-Minutes.md#decisions

**Review by samsonasik — APPROVED** [review]

**Pull request merged**
